File: Javascript/easy/Exercism/high-scores/high-scores.js
export class HighScores {
  #input;
  constructor(input) {
    this.#input = input;
  }

  get scores() {
    return this.#input;
  }

  get latest() {
    return this.#input[this.#input.length -1];
  }

  get personalBest() {
    return Math.max(...this.#input);
  }

  get personalTopThree() {
    if(this.#input.length === 1) return this.scores 

    this.#input.sort((a, b) => b - a);

    if(this.#input.length <= 3) return this.scores;

    return this.#input.slice(0,3);
  }
}

File: Java/easy/Exercism/high-scores/src/main/java/HighScores.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class HighScores {

    private List<Integer> scores;

    public HighScores(List<Integer> highScores) {
        this.scores = new ArrayList<>(highScores);
    }

    List<Integer> scores() {
        return this.scores;
    }

    Integer latest() {
        return this.scores.get(this.scores.size() - 1);
    }

    Integer personalBest() {
        return Collections.max(scores);
    }

    List<Integer> personalTopThree() {
        List<Integer> sortedScores = new ArrayList<>(scores);
        Collections.sort(sortedScores, Collections.reverseOrder());
        return sortedScores.subList(0, Math.min(3, sortedScores.size()));
    }

}
